### DISCUSSION TITLE
can i create a executable from meson package using pyinstaller

### DISCUSSION THREAD

**Discussion by MadadiSaeed** [mesonbuild/meson#9962] [Q&A] [answered]
If i create an executable from meson using pyinstaller, do i loose some of meson's functionalities?

**Comment by eli-schwartz** [comment, marked as answer]
That should work perfectly fine, Meson's own Windows installer uses pyinstaller to bundle Meson with python, ninja, pkg-config, etc.

Check out the `packaging/` directory.

**Reply by MadadiSaeed** [reply]
Thank you.

**Comment by ilg-ul** [comment]
As an alternate solution you can also use a binary distribution that already packs meson with an embedded python, and results in a standalone archive that has no system dependencies and can be installed in any location.

For details, please see the xPack Meson Build releases:

- <https://github.com/xpack-dev-tools/meson-build-xpack/releases>
